src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
+    contentWidth?: string | number;
+    /** The width of the area for the sidebar. */
+    sidebarWidth?: string | number;
+    /** If the sidebar is on the left or the right of the layout */
+    sidebarAlign?: SidebarAlignment;
+    /** The actual sidebar. This can be a component. */
+    sidebar?: any;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
     className?: string;
     children?: any;
 }
 
-export const Page = ({ children }: PageProps): React.ReactElement<any> => (
-    <StyledPageElement className="anchor-page">
+interface StyledLayoutProps {
+    layoutWidth?: string | number;
+    layoutBackgroundColor?: string;
+}
+
+interface StyledContentProps {
+    contentWidth?: string | number;
+    contentBackgroundColor?: string;
+}
+
+// Components
+// ------------------------------------------------------------------------------------------------------------------
+export const Page = (props: PageProps): React.ReactElement<any> => (
+    <StyledPage className={classnames('anchor-page', props.className)}>
         <NormalizeCSS />
         <GlobalCSS />
-        <Header />
-        <StyledBodyElement>{children}</StyledBodyElement>
-        <Footer />
-    </StyledPageElement>
+        <Grid
+            columns={1}
+            rows={`minmax(${props.headerHeight},auto) 1fr minmax(${
+                props.footerHeight
+            },auto)`}
+            height={'100vh'}
+        >
+            {props.header && <Cell>{props.header}</Cell>}
+
+            <Cell>{props.children}</Cell>
+
+            {props.footer && <Cell>{props.footer}</Cell>}
+        </Grid>
+    </StyledPage>
+);
+
+Page.defaultProps = PageDefaultProps;
+
+const DefaultLayout = (
+    props: DefaultLayoutProps
+): React.ReactElement<any> => (
+    <StyledLayout
+        layoutWidth={props.layoutWidth}
+        layoutBackgroundColor={props.layoutBackgroundColor}
+        className={classnames('anchor-default-layout', props.className)}
+    >
+        <StyledContent
+            contentWidth={props.contentWidth}
+            contentBackgroundColor={props.contentBackgroundColor}
+        >
+            {props.children}
+        </StyledContent>
+    </StyledLayout>
 );
+
+DefaultLayout.defaultProps = DefaultLayoutDefaultProps;
+
+const SidebarLayout = (
+    props: SidebarLayoutProps
+): React.ReactElement<any> => (
+    <StyledLayout
+        layoutWidth={props.layoutWidth}
+        layoutBackgroundColor={props.layoutBackgroundColor}
+        className={classnames('anchor-sidebar-layout', props.className)}
+    >
+        <StyledContent
+            contentWidth={props.contentWidth}
+            contentBackgroundColor={props.contentBackgroundColor}
+        >
+            {props.sidebarAlign === RIGHT ? (
```

Comment:
Ahh, typescript shows a little value here: we can use strict props to make binary configuration out of strings without having to have messy conditional blocks that check both sides of the condition.

Reply:
I'd like to talk about this more as I'm not sure how to go about that.